quickjspp.hpp

```diff
@@ -1963,6 +1991,49 @@ inline void Runtime::promise_unhandled_rejection_tracker(JSContext *ctx, JSValue
     }
 }
 
+inline JSModuleDef * Runtime::module_loader(JSContext *ctx,
+                                            const char *module_name, void *opaque)
+{
+    Context::ModuleData data;
+    auto & context = Context::get(ctx);
+
+    try {
+        if (context.moduleLoader) data = context.moduleLoader(module_name);
+
+        if (!data.source) {
+            JS_ThrowReferenceError(ctx, "could not load module filename '%s'", module_name);
+            return NULL;
+        }
+
+        if (!data.url) data.url = module_name;
```

Comment:
js_module_set_import_meta also prepends "file://", I wonder if it's required

Reply:
yes, it also resolves the canonical path (using `realpath`), which could be done using `#include <filesystem>`.

Reply:
Indeed, it also resolves the canonical path using `realpath`. That can be done here with `std::filesystem::weakly_canonical`.